catalog/opensearch/catalog-opensearch-endpoint/src/main/java/org/codice/ddf/opensearch/endpoint/query/OpenSearchQuery.java

```diff
@@ -408,6 +426,19 @@ public boolean evaluate(Object object) {
     return filter != null && filter.evaluate(object);
   }
 
+  public Query newInstanceWithFilter(Filter newFilter) {
+    OpenSearchQuery newQuery =
+        new OpenSearchQuery(
+            this.startIndex,
+            this.count,
+            this.sortBy,
+            this.maxTimeout,
+            this.siteIds,
+            this.filterBuilder);
+    newQuery.filters = Arrays.asList(newFilter);
```

Comment:
❓ I'm not very familiar with how opensearch queries work. Do I need to distinguish between `this.filters` and `this.spatialFilters`? It seems like they just end up getting concatenated together when they are used.